api/src/main/java/org/apache/cloudstack/api/command/admin/host/AddHostCmd.java

```diff
@@ -60,6 +60,9 @@
     @Parameter(name = ApiConstants.PASSWORD, type = CommandType.STRING, description = "the password for the host; required to be passed for hypervisors other than VMWare")
     private String password;
 
+    @Parameter(name = ApiConstants.PRIVATE_KEY, type = CommandType.STRING, description = "the private key for the host; support KVM only.", length = 16384)
+    private String privateKey;
```

Comment:
No this beats the purpose, we should assume that admin will setup and sync mgmt server ssh key for the cloud user across hosts.